tools/android/generate_breakpad_symbols.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+
+"""A tool to generate symbols for a binary suitable for breakpad.
+
+Wrapper for components/crash/content/tools/generate_breakpad_symbols.py for
+Android which takes the apk or aab and generates symbols for all libs
+"""
+
+import glob
+import multiprocessing
+import optparse
+import os
+import Queue
+import re
+import sys
+import subprocess
+import threading
+import zipfile
+
+CONCURRENT_TASKS=multiprocessing.cpu_count()
+
+def GetDumpSymsBinary(build_dir=None):
+  """Returns the path to the dump_syms binary."""
+  DUMP_SYMS = 'dump_syms'
+  dump_syms_bin = os.path.join(os.path.expanduser(build_dir), DUMP_SYMS)
+  if not os.access(dump_syms_bin, os.X_OK):
+    print 'Cannot find %s.' % dump_syms_bin
+    return None
+
+  return dump_syms_bin
+
+def InvokeChromiumGenerateSymbols(options, lib_paths):
+    """Invokes Chromium's script
+    components/crash/content/tools/generate_breakpad_symbols.py for each lib
+    of lib_paths."""
+
+    queue = Queue.Queue()
+    print_lock = threading.Lock()
+
+    at_least_one_failed = multiprocessing.Value('b', False)
+
+    chromium_script = os.path.join(options.src_root, 'components/crash/content/tools/generate_breakpad_symbols.py')
+
+
+    def _Worker():
+        while True:
+            lib_path = queue.get()
+
+            try:
+                # Invoke the original Chromium script
+                args = ['python',
+                        chromium_script,
+                        '--build-dir=' + options.build_dir,
+                        '--symbols-dir=' + options.symbols_dir,
+                        '--binary=' + lib_path,
+                        '--platform=android',
+                        '--verbose'
+                        ]
+
+                ret = subprocess.call(args)
+
+                if (ret != 0):
+                    # Lets fail just not to ignore something important
+                    at_least_one_failed.value = True
+
+            except Exception, e: # pylint: disable=broad-except
+                if options.verbose:
+                    with print_lock:
+                        print(type(e))
+                        print(e)
+            finally:
+                queue.task_done()
+
+    for lib_path in lib_paths:
+        queue.put(lib_path)
+
+    for _ in range(options.jobs):
+        t = threading.Thread(target=_Worker)
+        t.daemon = True
+        t.start()
+
+    queue.join()
+
+    if at_least_one_failed.value:
+        return 1
+
+    return 0
+
+def main():
+    parser = optparse.OptionParser()
+    parser.add_option('', '--build-dir', default='',
+                    help='The build output directory.')
+    parser.add_option('', '--symbols-dir', default='',
+                    help='The directory where to write the symbols file.')
+    parser.add_option('', '--apk-or-aab', default='',
```

Comment:
I think the name is a little confusing because it sounds like a switch, maybe call it `--binray` or something like that?

Reply:
@wchen342 you are right, because it is already used in build script and supposed options are: `--target_android_output_format=apk` or `--target_android_output_format=apk`. I will change this name.

Reply:
fixed in https://github.com/brave/brave-core/pull/9665/commits/18d2717e72afb427aef8056349a22c3a35401ebb